VideoLocker/src/main/java/org/edx/mobile/view/adapters/MyCourseAdapter.java

```diff
@@ -104,10 +107,11 @@ public void onClick(View v) {
                 Date startDate = DateUtil.convertToDate(courseData.getStart());
                 String startDt; 
                 Date endDate = DateUtil.convertToDate(courseData.getEnd());
-                String endDt; 
-
-                if (!courseData.isStarted()) {
-                    if(startDate!=null){
+                String endDt;
+                AccessError error = enrollment.getCourse().getCourseware_access().getError_code();
+                if (error.equals(AccessError.StartDateError)) {
```

Comment:
It seems like there are more null safe ways to compare enums.  http://stackoverflow.com/questions/1750435/comparing-java-enum-members-or-equals has a good discussion.